feat: Added APIGatewayMethod ingestion and relationship

Khushmagrawal opened the pull request:
### Summary
This PR adds ingestion and relationships for AWS API Gateway Methods. Relationships include:
- (:APIGatewayResource)-[:RESOURCE]->(:APIGatewayMethod) 
- (:APIGatewayMethod)-[:INVOKES]->(:AWSLambda) 
- (:APIGatewayMethod)-[:ACCESSES]->(:S3Bucket) 
- (:APIGatewayMethod)-[:ACCESSES]->(:DynamoDBTable) 



### Related issues or links
( #1552 )


### Checklist

Provide proof that this works (this makes reviews move faster). Please perform one or more of the following:
- [x] Update/add unit or integration tests.
- [ ] Include a screenshot showing what the graph looked like before and after your changes.
- [ ] Include console log trace showing what happened before and after your changes.

If you are changing a node or relationship:
- [x] Update the [schema](https://github.com/cartography-cncf/cartography/tree/master/docs/root/modules) and [readme](https://github.com/cartography-cncf/cartography/blob/master/docs/schema/README.md).

If you are implementing a new intel module:
- [x] Use the NodeSchema [data model](https://cartography-cncf.github.io/cartography/dev/writing-intel-modules.html#defining-a-node).

jychp reviewed (changes requested):
`sub_resource` (with label `RESOURCE`) should represent the relationship between the tenant (in this case, the `AWSAccount`) and the node, whenever applicable.

In this case, you should:

- [ ] Rename this relationship to `CONTAINS` (or use a more appropriate type, if needed)
- [ ] Move this relationship to `other_relationships`
- [ ] Create a new relationship pointing to the `AWSAccount` node
- [ ] Set this new relationship as the `sub_resource` relationship

Khushmagrawal commented:
> `sub_resource` (with label `RESOURCE`) should represent the relationship between the tenant (in this case, the `AWSAccount`) and the node, whenever applicable.
> 
> In this case, you should:
> 
> * [ ]  Rename this relationship to `CONTAINS` (or use a more appropriate type, if needed)
> * [ ]  Move this relationship to `other_relationships`
> * [ ]  Create a new relationship pointing to the `AWSAccount` node
> * [ ]  Set this new relationship as the `sub_resource` relationship

Oh yes, it should always represent the higher level parent so added and corrected the relationships, also updated the `cleanup()` (maintaining the order) and updated tests accordingly.

Khushmagrawal commented:
Also, there can be a relationship from APIGatewayMethod to AWSVpc via APIGatewayVpcLink, which would look like this:
(:APIGatewayMethod)-[:CONNECTS_VIA]->(:APIGatewayVpcLink)-[:CONNECTS_TO]->(:AWSVpc)
However, this requires first defining the `APIGatewayVpcLink` ([AWS_doc])(https://docs.aws.amazon.com/apigateway/latest/api/API_VpcLink.html)

I am planning to implement it in a future PR to keep this one focused.

jychp commented on cartography/intel/aws/apigateway.py:
```diff
@@ -239,6 +390,8 @@ def transform_rest_api_details(
         if len(resource) > 0:
             for r in resource:
                 r["apiId"] = api_id
+                r["region"] = region
```
Why did you add this in the transformation ?

Region & Account ID are available as kwargs and are not needed to be set in the dict, right ?

jychp reviewed (changes requested):
Thanks for your contribution!
I think there are several places in your code where things could be simplified — and simpler is always better 🙂

If you're using AI tools like GitHub Copilot or Cody (which is totally fine, we're strong advocates of using AI), just make sure to provide enough context and guidance. For example, you can explicitly ask to keep things simple or reference existing files as examples to align with the project's conventions.

achantavy commented:
Hey @Khushmagrawal, are you available to follow up on the review comments? Totally get it if not, things get busy, just let us know

Khushmagrawal commented:
Hey @achantavy,  I'm currently tied up with some other commitments and likely won’t be able to address the review comments for another month. Would it make sense to close the PR for now and revisit it later? happy to follow whatever works best.

Khushmagrawal commented:
Closing this as #1819 implemented this. Thanks @d-aggarwal
